odev.js

```diff
@@ -31,6 +31,55 @@ document.querySelector("#reset").addEventListener("click", () => {
     createTableElements([], "singlecity")
 });
 
+ 
 /* START CODING HERE */
 
+document.querySelector("#populationBigger").addEventListener("click", () => {
+  let popBigger = data.filter(elements => elements.population > 500000)
+   createTableElements(popBigger,"allcities");
+})
 
+document.querySelector("#landAreaLess").addEventListener("click",()=> {
+  let lessThan = data.filter(elements => elements.landArea < 1000)
+   createTableElements(lessThan,"allcities");
+})
+
+document.querySelector("#isPopulationLess").addEventListener("click",() => {
+  let populationSome = data.some(elements => elements.population < 100000)
+  if(populationSome){
+    alert("YES")
+  }
+  else {
+    alert ("NO")
+  }
+})
+
+document.querySelector("#isLandBigger").addEventListener("click",()=> {
+  let landEvery = data.every(elements => elements.landArea > 100)
+  if(landEvery){
+    alert("YES")
+  }
+  else{
+    alert("NO")
+  }
+})
+
+const dataCities = data.map(cities => cities.name);
+  const citySelect =document.querySelector("#inputGroupSelect01")
+  for(let i = 0; i<=citySelect.length; i++){
```

Comment:
elemanlarda for donup silmek yerine innerHTML = "" diyerek icini bosaltabilirdin. Dom islemleri biraz maliyetli islemler, gereksiz olanlardan kacinmamiz gerekiyor.

Reply:
Teşekkür ederim hocam daha dikkat edeceğim.